eox_tagging/api/v1/test/test_viewset.py

```diff
@@ -137,43 +157,117 @@ def test_create_tag_with_wrong_datetime_format(self):
 
         self.assertEqual(response.status_code, 400)
 
-    def test_patch_tag(self):
+    @patch_permissions
+    def test_create_tag_with_owner_site(self, _):
+        """"Used to test creating a tag with site as owner."""
+
+        data = {
+            "tag_type": "example_tag_2",
+            "tag_value": "example_tag_value",
+            "target_type": "user",
+            "target_id": "user_test",
+            "owner_type": "site",
+            "access": "PRIVATE",
+            "expiration_date": "2020-12-04 10:30:40",
+        }
+
+        response = self.client.post(self.URL, data, format='json')
+
+        self.assertEqual(response.data["owner_type"].lower(), "site")
+
+    @patch_permissions
+    def test_create_tag_with_wrong_owner(self, _):
+        """"Used to test creating a tag with wrong owner_type. This results in bad request."""
+        data = {
+            "tag_type": "example_tag_1",
+            "tag_value": "example_tag_value",
+            "target_type": "user",
+            "target_id": "user_test",
+            "owner_type": "course",  # default is site
+            "access": "PRIVATE",
+            "expiration_date": "2020-12-04 10:30:40 ",
+        }
+
+        response = self.client.post(self.URL, data, format='json')
+
+        self.assertEqual(response.status_code, 400)
+
+    @patch_permissions
+    def test_patch_tag(self, _):
         """Used to test that a tag can't be updated."""
         response = self.client.patch("{URL}{key}/".format(URL=self.URL, key=self.example_tag.key.hex))
 
         self.assertEqual(response.status_code, 405)
 
-    def test_put_tag(self):
+    @patch_permissions
+    def test_put_tag(self, _):
         """Used to test that a tag can't be updated."""
         response = self.client.put("{URL}{key}/".format(URL=self.URL, key=self.example_tag.key.hex))
 
         self.assertEqual(response.status_code, 405)
 
-    def test_filter_by_tag_key(self):
+    @patch_permissions
+    def test_filter_by_tag_key(self, _):
         """Used to test getting a tag given its key."""
-        url = "{URL}?key={key}".format(URL=self.URL, key=self.example_tag.key.hex)
+        URL = "{URL}?key={key}".format(URL=self.URL, key=self.example_tag.key.hex)
```

Comment:
define the query params as data and use it in self.client.get(self.url , data=data) instead of create the url

Reply:
nice. Done